src/Raven.Server/Utils/BackupUtils.cs

```diff
@@ -531,12 +542,57 @@ private static IdleDatabaseActivity GetNextIdleDatabaseActivity(NextIdleDatabase
         return new IdleDatabaseActivity(IdleDatabaseActivityType.UpdateBackupStatusOnly, nextBackup.DateTime, parameters.Configuration.TaskId, parameters.LastEtag);
     }
 
+    public static PeriodicBackupStatus GetLocalBackupStatus(ServerStore serverStore, string databaseName, long taskId)
+    {
+        using (serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+        using (context.OpenReadTransaction())
+        {
+            return GetLocalBackupStatus(serverStore, context, databaseName, taskId);
+        }
+    }
+
+    public static PeriodicBackupStatus GetLocalBackupStatus(ServerStore serverStore, TransactionOperationContext context, string databaseName, long taskId)
+    {
+        var localStatus = serverStore.DatabaseInfoCache.BackupStatusStorage.GetBackupStatus(databaseName, serverStore._env.Base64Id, taskId, context);
+        if (localStatus != null)
+            return localStatus;
+
+        // backwards compatibility - fallback to cluster status - only if node tag matches
+        var clusterStatus = GetBackupStatusFromCluster(serverStore, context, databaseName, taskId);
+
+        if (clusterStatus == null || clusterStatus.NodeTag != serverStore.NodeTag)
+            return null;
+
+        return clusterStatus;
+    }
+
+    public static BlittableJsonReaderObject GetLocalBackupStatusBlittable<T>(ServerStore serverStore, TransactionOperationContext<T> context, string databaseName, long taskId) where T : RavenTransaction
+    {
+        var localStatus = serverStore.DatabaseInfoCache.BackupStatusStorage.GetBackupStatusBlittable(context, databaseName, serverStore._env.Base64Id, taskId);
+        if (localStatus != null)
+            return localStatus;
+
+        // backwards compatibility - fallback to cluster status - only if node tag matches
+        var clusterStatus = GetBackupStatusFromClusterBlittable(serverStore, context, databaseName, taskId);
+
+        if (clusterStatus == null || clusterStatus.TryGet(nameof(PeriodicBackupStatus.NodeTag), out string nodeTag) == false || string.IsNullOrEmpty(nodeTag))
+            return null;
+
+        if (nodeTag == serverStore.NodeTag)
+            return clusterStatus;
+
+        return null;
+    }
+
     public static void SaveBackupStatus(PeriodicBackupStatus status, string databaseName, ServerStore serverStore, Logger logger,
         BackupResult backupResult = default, Action<IOperationProgress> onProgress = default, OperationCancelToken operationCancelToken = default)
     {
         try
         {
             var raftId = RaftIdGenerator.NewId();
+            // update backup status locally
+            serverStore.DatabaseInfoCache.BackupStatusStorage.InsertBackupStatus(status, databaseName, serverStore._env.Base64Id, status.TaskId);
```

Comment:
Since we need to open a write transaction here, can we avoid it and do this only in case we were unable to send the command to the cluster?
Note that we need to save the status only on our server.

Reply:
Done